### DISCUSSION TITLE
Any way to know that the project doesn't contain any malware?

### DISCUSSION THREAD

**Discussion by AaronWNW** [mrkai77/Loop#376] [Q&A]
Hey Loop developers,

I really like the project and recommended it to my dad. He uses Apple products mainly because of the safety. He also liked the project but asked me if there is any way to know that the project is save. Obviously the source code is public but that's not a guarantee for somebody who doesn't want to read and check all the code.
I was just curious if the is any certificate or any other guarantee that the project is save to install. If not are there any plans to do so, like releasing it in the App-Store?

Best regards,
Aaron W.

**Comment by mrkai77** [comment]
No problem, this is a totally valid question. 

Since the project is completely open-source, it can be checked if it has any malicious code. And if you believe that there's a possibility that the GitHub releases don't fully reflect the source code, you can also fully compile the full app yourself (I can also help you with that) :)

Unfortunately, Loop is not eligible to be put onto the App Store, since it uses the Accessibility API to move other processes' windows. Other window managers such as Magnet were submitted to the App Store *before* Apple introduced this restriction, which is why they are still available on the App Store.

For the certificate, that requires an Apple Developer Program subscription, which I finally managed to get around a week ago. This means that the next release will be fully notarized by Apple, which means that they would have verified that Loop doesn't contain any malicious code.

Please feel free to ask more regarding security  - I am fully committed to keeping Loop malware-free!

**Reply by AaronWNW** [reply]
Thanks for the answer 😊

Nice to hear that you have plans to get the certificate with the developer subscription :) 
I’ll recommend Loop to my dad again at the next release with the certificate. I think he will be happy to install it. 

Keep up the great work 👍

**Reply by mrkai77** [reply]
Hey @AaronWNW! 1.0.0 has now been released, and is notarized :D

<img width="841" alt="image" src="https://github.com/MrKai77/Loop/assets/68963405/313f5a32-ffcd-449a-9372-d86c04608fc6">
